bin/bump-version

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+usage() { echo "Usage: $0 -p [major | minor | patch]" 1>&2; exit 1; }
+
+while getopts "p:" o; do
+  case "${o}" in
+    p)
+      patch_level=${OPTARG}
+      (( patch_level == 'major' || patch_level == 'minor' || patch_level == 'patch'))
```

Comment:
I wasn't familiar with the [double parentheses construct](https://tldp.org/LDP/abs/html/dblparens.html) before this 😄

Could you explain to me what effect this has on the script? It _look_ like a check to see whether the `patch_level` is supported, but I'm having a hard time reasoning about the impact of the check on the script.

Reply:
This is really just a defensive catch on avoiding passing a garbage argument into NPM. It also acts as a filter on arguments we don't use/permit, like prepatch.

It does mean we also prevent you actually specifying the version number to use by hand, but I consider that kind of YAGNI as in that scenario you are kind of off-script on how we work so the rest of the automated steps maybe don't help.